### DISCUSSION TITLE
pytester and plugins using `pytest_configure`

### DISCUSSION THREAD

**Discussion by The-Compiler** [pytest-dev/pytest#10583] [Q&A]
I'm using [pytest-golden](https://github.com/oprypin/pytest-golden) as well as pytester in the same project. pytest-golden happens to [define a `pytest_configure` hook](https://github.com/oprypin/pytest-golden/blob/a45058da8ef5dfffef3b3586ae5ecd6a92f0c7a0/pytest_golden/plugin.py#L52-L56), which warns about `enable_assertion_pass_hook` being unset, if that's the case.

This is problematic in combination with pytester, because in the pytester run, the plugin gets loaded, but I'd rather not have to define a dummy .ini file to get rid of the warning. However, I don't actually use golden in those pytester tests... Here's a minimal example:

pytest.ini:

```ini
[pytest]
enable_assertion_pass_hook=true
```

basic.yml:

```yaml
input: abc
output: 3
```

test_golden.py:

```python
import pytest

pytest_plugins = "pytester"

@pytest.mark.golden_test("basic.yml")
def test_bar(golden):
    assert len(golden["input"]) == golden.out["output"]


def test_pytester(pytester):
    pytester.makepyfile("def test_pytester(): pass")
    pytester.runpytest()
```

This leads to:

```
============================= test session starts ==============================
platform linux -- Python 3.10.8, pytest-7.2.0, pluggy-1.0.0
rootdir: /home/florian/tmp/golden, configfile: pytest.ini
plugins: golden-0.2.2
collected 2 items                                                              

test_golden.py ..                                                        [100%]

=============================== warnings summary ===============================
test_golden.py::test_pytester
  /home/florian/tmp/golden/.venv/lib/python3.10/site-packages/pytest_golden/plugin.py:53: GoldenTestUsageWarning: Add 'enable_assertion_pass_hook=true' to pytest.ini for safer usage of pytest-golden.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================= 2 passed, 1 warning in 0.04s =========================
```

How do I best deal with this? Is there something the pytest-golden plugin should change to make this easier? I've tried using `pytester.runpytest(plugins=["no:golden"])`, or `pytester.runpytest("-p", "no:golden")`, but that doesn't seem to change anything for an inline run.

**Comment by ckutlu** [comment]
I just had the same question. Looks like there was some discussion going on about this in https://github.com/pytest-dev/pytest/issues/13369.
